tests/TypehintTest.hack

```diff
@@ -22,7 +22,25 @@ final class TypehintTest extends \Facebook\HackTest\HackTest {
       tuple('(string, string,)', 'tuple', '(string,string)'),
       tuple('(function(): void)', 'callable', '(function():void)'),
       tuple('(function(string,): int)', 'callable', '(function(string):int)'),
-      tuple('(function(a,b): int)', 'callable', '(function(a,b):int)'),
+      tuple(
+        '(function(a,b): int)',
+        'callable',
+        '(function(MyNamespace\\a,MyNamespace\\b):int)',
+      ),
+
+      // Shape with a namespaced field
+      tuple(
+        'shape("foo" => string, "bar" => Baz)',
+        'shape',
+        'shape("foo"=>string,"bar"=>MyNamespace\\Baz)',
+      ),
+
+      // Function with an inout param
+      tuple(
+        '(function(inout Foo): Bar)',
+        'callable',
+        '(function(inout MyNamespace\\Foo):MyNamespace\\Bar)',
```

Comment:
Something weird's going on I don't understand: why is there no space after `:` here, but there is at the examples above?

If I'm reading correctly, it's always `':'.$return_type->getTypeText()`, so rather than just being a nit, this implies we're getting a bogus space in the return type type text

Reply:
The strings with spaces are inputs, all the outputs are without space (we test that the output has consistent formatting no matter how the input is formatted).

I tried to preserve the original behavior of removing all "trivia".